src/main/java/net/minecraftforge/client/model/ItemLayerModel.java

```diff
@@ -429,13 +431,26 @@ private static void putVertex(UnpackedBakedQuad.Builder builder, VertexFormat fo
             case COLOR:
                 builder.put(e, 1f, 1f, 1f, 1f);
                 break;
-            case UV: if(format.getElement(e).getIndex() == 0)
-            {
-                builder.put(e, u, v, 0f, 1f);
+            case UV:
+                if(format.getElement(e).getIndex() == 0)
+                {
+                    builder.put(e, u, v, 0f, 1f);
+                }
                 break;
-            }
             case NORMAL:
-                builder.put(e, (float)side.getFrontOffsetX(), (float)side.getFrontOffsetY(), (float)side.getFrontOffsetZ(), 0f);
+                if(hasTransform)
+                {
+                    vec.x = (float)side.getFrontOffsetX();
+                    vec.y = (float)side.getFrontOffsetY();
+                    vec.z = (float)side.getFrontOffsetZ();
+                    vec.w = 1f;
+                    transform.get().getMatrix().transform(vec);
```

Comment:
```suggestion
                    Matrix4f mat = transform.map(TRSRTransformation::getMatrix).orElseGet(Matrix4f::new);
                    mat.invert();
                    mat.transpose();
                    mat.transform(vec);
                    vec.normalize();
```
@RainWarrior would this suffice?

Reply:
I'd expect this to be very slow, since the invertion/normalization happens for every vertex, but yes, this should be correct